backend/src/app.py

```diff
@@ -274,14 +432,23 @@ def account_type_valid(acct_type: int):
     return False
 
 
+def transaction_type_valid(category: int):
```

Comment:
non-blocking: If additional data types beyond `Account` and `Transaction` are created with similar query patterns, consider extracting the common validity check such as by:

```python
def is_valid(categories, category):
    if category is None:
        return False
    for valid_category in categories:
        if category == valid_category.value:
            return True
    return False


def account_type_valid(account):
    return is_valid(AccountType, account)


def transaction_type_valid(category):
    return is_valid(TransactionType, category)
```

Reply:
I see, added something similar just now that follows that example with converting list of rows into dicts instead